src/main.rs

```diff
@@ -869,4 +914,213 @@ mod tests {
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            minimum_abi = "1.0"
+            
+            [sgx]
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            "#;
+
+            let csr_object = CertReq::from_der(ICELAKE_CSR).unwrap();
+            let config_obj: ConfigurationFile =
+                toml::from_str(config_raw).expect("Couldn't deserialize");
+
+            if validate_sgx_config(&csr_object, &config_obj.sgx_config.unwrap()).is_ok() {
+                unreachable!();
+            }
+        }
+
+        #[test]
+        fn test_sgx_signed_csr_bad_config_enclave_version() {
+            let config_raw = r#"
+            [snp]
+            policy_flags = "SingleSocket | Debug"
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            minimum_abi = "1.0"
+            
+            [sgx]
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            enclave_security_version = 9999
+            "#;
+
+            let csr_object = CertReq::from_der(ICELAKE_CSR).unwrap();
+            let config_obj: ConfigurationFile =
+                toml::from_str(config_raw).expect("Couldn't deserialize");
+
+            if validate_sgx_config(&csr_object, &config_obj.sgx_config.unwrap()).is_ok() {
+                unreachable!();
+            }
+        }
+
+        #[test]
+        fn test_snp_signed_canned_csr() {
+            let csr_object = CertReq::from_der(MILAN_CSR).unwrap();
+            let config_obj: ConfigurationFile =
+                toml::from_str(STEWARD_CONFIG).expect("Couldn't deserialize");
+            validate_snp_config(&csr_object, &config_obj.snp_config.unwrap()).unwrap();
+        }
+
+        #[test]
+        fn test_snp_signed_canned_csr_bad_author_key() {
+            let csr_object = CertReq::from_der(MILAN_CSR).unwrap();
+
+            let config_raw = r#"
+            [snp]
+            policy_flags = "SingleSocket | Debug"
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            
+            [sgx]
+            enclave_security_version = 9999
+            enarx_signer = ["1234567890", "00112233445566778899"]
+            "#;
+            let config_obj: ConfigurationFile =
+                toml::from_str(config_raw).expect("Couldn't deserialize");
+
+            if validate_snp_config(&csr_object, &config_obj.snp_config.unwrap()).is_ok() {
+                unreachable!();
+            }
+        }
+
+        #[test]
+        fn test_snp_signed_canned_csr_bad_abi_version() {
+            let csr_object = CertReq::from_der(MILAN_CSR).unwrap();
+
+            let config_raw = r#"
+            [snp]
+            enarx_signer = ["c8dc9fe36caaeef871e6512c481092754c57c2ea999f128282ccb563d1602774"]
+            minimum_abi = "254.0"
+            "#;
+            let config_obj: ConfigurationFile =
+                toml::from_str(config_raw).expect("Couldn't deserialize");
```

Comment:
Do we gain anything this way over just doing 
```rust
let config = toml::from_str(r#"
[snp]
enarx_signer = ["c8dc9fe36caaeef871e6512c481092754c57c2ea999f128282ccb563d1602774"]
minimum_abi = "254.0"
"#).expect("failed to deserialize config")
```

Reply:
Fixed